src/GovNotifyHandler.ts

```diff
@@ -41,26 +42,45 @@ class GovNotifyHandler implements LambdaInterface {
 					logger.info({ message: "Fetching GOVUKNOTIFY_API_KEY from SSM" });
 					try {
 						GOVUKNOTIFY_API_KEY = await getParameter(this.environmentVariables.govNotifyApiKeySsmPath());
-					} catch (err) {
-						logger.error(`failed to get param from ssm at ${this.environmentVariables.govNotifyApiKeySsmPath()}`, { err });
-						throw err;
+					} catch (error) {
+						logger.error(`failed to get param from ssm at ${this.environmentVariables.govNotifyApiKeySsmPath()}`, { error });
+						batchFailures.push({ itemIdentifier: "" });
+						return { batchItemFailures: batchFailures };
 					}
 				}
 				let govnotifyServiceId;
 				try {
 					govnotifyServiceId = GOVUKNOTIFY_API_KEY.substring(GOVUKNOTIFY_API_KEY.length - 73, GOVUKNOTIFY_API_KEY.length - 37);
-				}catch (error) {
+				} catch (error) {
 					logger.error("failed to extract govnotifyServiceId from the GOVUKNOTIFY_API_KEY", { error });
 					batchFailures.push({ itemIdentifier: "" });
 					return { batchItemFailures: batchFailures };
 				}
-				await SendEmailProcessor.getInstance(logger, metrics, GOVUKNOTIFY_API_KEY, govnotifyServiceId, this.environmentVariables.sessionEventsTable()).processRequest(body);
+				// Check for messageType and reject it if it doesnt match the required types.
+				const messageType = body.Message.messageType;
+				let message;
+				switch (messageType) {
+					case Constants.OLD_EMAIL: {
```

Comment:
With Reminder email being introduced and possibly others in future could we rename OLD and NEW_EMAIl to be something more meaningful 🙏